src/main/java/com/ecore/roles/client/UsersClient.java

```diff
@@ -0,0 +1,37 @@
+package com.ecore.roles.client;
+
+import com.ecore.roles.client.model.User;
+import com.ecore.roles.configuration.ClientsConfigurationProperties;
+import lombok.RequiredArgsConstructor;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+import java.util.UUID;
+
+@RequiredArgsConstructor
```

Comment:
**Not a must**
For sure it helps with getting less code, but in my opinion this is a case where it's better have the constructor with the Autowired annotation. It makes the code more readable in spring beans. Once we have both standards in the current code, I suggest to go just with one of them as much as possible.